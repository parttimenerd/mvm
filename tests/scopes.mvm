PUSH_INT 3
SET_VAR abc
PUSH_VAR abc
PUSH_VAR println
CALL_N 1
POP
PUSH_SCOPE
PUSH_INT 4
SET_VAR abcd
PUSH_VAR abcd
PUSH_VAR println
CALL_N 1
PUSH_VAR abc
PUSH_VAR println
CALL_N 1
PUSH_INT 22
SET_VAR_HERE abc
PUSH_VAR abc
PUSH_VAR println
CALL_N 1 
POP
POP_SCOPE
PUSH_VAR abcd
PUSH_VAR println
CALL_N 1
PUSH_VAR abc
PUSH_VAR println
CALL_N 1
