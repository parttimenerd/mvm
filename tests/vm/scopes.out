3
4
3
22
nothing
3

